### DISCUSSION TITLE
请问 czg 的配置是否可以编写于 commitlint.config.js 中

### DISCUSSION THREAD

**Discussion by jackblackevo** [Zhengqbbb/cz-git#126] [Q&A] [answered]
您好！
不好意思，新手有一些不清楚的地方想询问 🙇‍♂️

文档中写到，cz-git 可以在 commitlint.config.js 中进行配置。
请问 czg 是否也同样能够将编写配置于 commitlint.config.js 中呢？

也想请问 cz-git 与 commitlint 的关系是什么呢？
我目前的理解是，commitlint 会校验 git commit message 格式；而 cz-git 则是提供了 commit 讯息格式，辅助用户生成 commit 讯息。

cz-git 的配置撰写于 commitlint.config.js 中，对于 commitlint 有什么影响吗？
而 cz-git 会从 commitlint 获取什么呢？

谢谢您 🙏

**Comment by Zhengqbbb** [comment, marked as answer]
很抱歉，回复晚了

---

 > 请问 czg 是否也同样能够将编写配置于 commitlint.config.js 中呢？

czg 和 cz-git 都采用一套配置加载，所以可以将配置写于 `commitlint`

---

> 请问 cz-git 与 commitlint 的关系是什么呢？| cz-git 会从 commitlint 获取什么呢？

和 commitlint 没有关系，但是配置加载会加载 `commitlint` 的规则配置信息，并将其提供到 CLI (比如最大讯息长度等等)

---

> 我目前的理解是，commitlint 会校验 git commit message 格式；而 cz-git 则是提供了 commit 讯息格式，辅助用户生成 commit 讯息

是的

---

> cz-git 的配置撰写于 commitlint.config.js 中，对于 commitlint 有什么影响吗？

对于 `commitlint` 不会有影响，我们的配置都是 wrap 了一个 `prompt` Object
```js
module.exports = {
  rules: {
 
  },
 prompt: {
  //配置编写于这里
 }
}
```

---
感谢你的支持，如果有其他疑问可以回复 ～ 🤗

**Reply by jackblackevo** [reply]
谢谢您！
我也试着看了看 code～

- cz-git 的 [`clLoader`](https://github.com/Zhengqbbb/cz-git/blob/1ef487f65b9211ad0f1c7f9db34ff9f1a8b9ee4a/packages/%40cz-git/plugin-loader/src/index.ts#L69) 会去加载 commitlint 配置信息，并且取得配置中的 `rules` 和 `prompt`。
- 而 czg 又会去呼叫 cz-git 的 [`prompter`](https://github.com/Zhengqbbb/cz-git/blob/1ef487f65b9211ad0f1c7f9db34ff9f1a8b9ee4a/packages/cli/src/generator/czg.ts#L44)，所以 czg 也会从 commitlint.config.js 去获取规则配置信息还有 cz-git 自己的配置信息。

不知道这样理解对不对 🙂

**Reply by Zhengqbbb** [reply]
是的
